controllers/metal3.io/host_state_machine.go

```diff
@@ -417,6 +417,13 @@ func (hsm *hostStateMachine) handleAvailable(info *reconcileInfo) actionResult {
 	} else if dirty {
 		hsm.NextState = metal3v1alpha1.StatePreparing
 		return actionComplete{}
+	} else {
+		// Check if hostFirmwareSettings have been changed
+		dirty, _, _ := hsm.Reconciler.getHostFirmwareSettings(info)
+		if dirty {
+			hsm.NextState = metal3v1alpha1.StatePreparing
```

Comment:
@bfournie it would be nice to have some coverage on this part (considering that a state change is a relevant action). As a good starting point for that you could have a look at the [baremetalhost controller test](https://github.com/metal3-io/baremetal-operator/blob/ea7afff67eefb9360cc278dc4f34e340461551f4/controllers/metal3.io/baremetalhost_controller_test.go)

Reply:
Good point, added a test for getHostFirmwareSettings()